src/main/java/nl/tudelft/lifetiles/traverser/models/EmptySegmentTraverser.java

```diff
@@ -0,0 +1,185 @@
+package nl.tudelft.lifetiles.traverser.models;
+
+import java.util.HashSet;
+import java.util.PriorityQueue;
+import java.util.Set;
+
+import nl.tudelft.lifetiles.graph.models.Edge;
+import nl.tudelft.lifetiles.graph.models.Graph;
+import nl.tudelft.lifetiles.graph.models.sequence.SegmentEmpty;
+import nl.tudelft.lifetiles.graph.models.sequence.Sequence;
+import nl.tudelft.lifetiles.graph.models.sequence.SequenceSegment;
+
+/**
+ * Places empty vertices in the graph where there are unified gaps.
+ *
+ * @author Jos
+ *
+ */
+public class EmptySegmentTraverser {
+    /**
+     * Traverser's copy of the graph.
+     */
+    private Graph<SequenceSegment> graphVar;
+
+    /**
+     * Traverses the copied graph.
+     *
+     * @param graph
+     *            the graph to copy and traverse.
+     * @return the traversed copy of the graph.
+     */
+    public final Graph<SequenceSegment> traverseGraph(
+            final Graph<SequenceSegment> graph) {
+        graphVar = graph;
+        traverseGraph();
+        return this.graphVar;
+    }
+
+    /**
+     * Traverses the copied graph.
+     */
+    public final void traverseGraph() {
+        for (SequenceSegment vertex : new HashSet<SequenceSegment>(
```

Comment:
Creating a `List` is quicker than creating a `Set` because of the uniqueness checks (although `O(1)`) and creating a set here does not have additional value in comparison to creating a list.